src/api/services/booksService.ts

```diff
@@ -11,29 +11,29 @@ import { messages } from '@/sources/messages';
 import { isApiErrorResponse } from '@/utils/isApiErrorResponse';
 
 export const bookService = {
-  getBooksList: async (query: string): Promise<BookData[]> => {
+  getBooksList: async (
+    query: string,
+    page: number,
+    pageItemsResults: number
+  ): Promise<{ books: BookData[]; totalItems: number }> => {
     const trimmedQuery = query.trim();
+    const startIndex = (page - 1) * pageItemsResults;
 
     try {
       const searchQuery = trimmedQuery ? `intitle:${query.trim()}` : 'book';
 
       const response: AxiosResponse<IBooksListResponse> = await booksApi.get(
-        `?q=${encodeURIComponent(searchQuery)}&maxResults=20&langRestrict=en&key=${BOOKS_API_KEY}`
+        `?q=${encodeURIComponent(searchQuery)}&startIndex=${startIndex}&maxResults=${pageItemsResults}&langRestrict=en&key=${BOOKS_API_KEY}`
       );
+
       const booksResult: IBookItemResponse[] = Array.isArray(
         response.data.items
       )
         ? response.data.items
         : [];
```

Comment:
эта штука вызывает вопросы, потому что у тебя тип респонса описан не совсем верно

<img width="836" height="256" alt="image" src="https://github.com/user-attachments/assets/f2083ae9-2e78-4761-be4c-30cf2e004521" />

у тебя тут items - обязательный параметр. но если поиск ничего не находит, то ответ приходит такой

<img width="814" height="278" alt="image" src="https://github.com/user-attachments/assets/3649cc30-b3fb-4b94-8e51-91c5aa20f639" />

как видишь, items там нет. в таком случае, если сделать поле опциональным, то эту штуку можно упростить до

```suggestion
      const booksResult = response.data.items || [];
```
оно у тебя и так работает, но вот так будет понятно что происходит